extensions/legit/legit.lisp

```diff
@@ -538,11 +545,74 @@ Currently Git-only. Concretely, this calls Git with the -w option.")
                            ;; Save the hash on this line for later use.
                            (when hash
                              (put-text-property start point :commit-hash hash)))))
-              (lem/peek-legit:collector-insert "<none>")))
+              (lem/peek-legit:collector-insert collector "<none>")))
 
         (add-hook (variable-value 'after-change-functions :buffer (lem/peek-legit:collector-buffer collector))
                   'change-grep-buffer)))))
 
+(defun get-clgit-walker (path)
+  (arrow-macros:-> path
+    lem-core/commands/project:find-root
+    cl-git:open-repository
+    cl-git:repository-head
+    cl-git:target
+    (lambda (x) 
+      (cl-git:revision-walk x :ordering :topological))))
+
+(define-command legit-log () ()
+  "Show log buffer."
+  (with-current-project ()
+    (lem/peek-legit:with-collecting-sources (collector :read-only nil)
+      (if (not (lem/porcelain:git-project-p))
+          (lem/peek-legit:collector-insert collector "[Error] Only works for git repos" :header t)
+          (progn
+            (lem/peek-legit:collector-insert collector "[WARNING] Branching history is NOT shown well" :header t)
+
+            ;; Header: current branch.
+            (lem/peek-legit:collector-insert collector
+                                             (format nil "Branch: ~a" (lem/porcelain:current-branch))
+                                             :header t)
+            (lem/peek-legit:collector-insert collector "")
+
+            ;; Commits.
+            (let* ((entry-group-size 20) ;; TODO(Magic number) Add parameter to control number?
+                   (shorthash-size 12)   ;; TODO(Magic number) Add parameter to control hash length?
+                   (walker (get-clgit-walker (buffer-directory)))
+                   (buffer (lem/peek-legit:collector-buffer collector)))
+              (labels ((add-log-entries () 
+                         (loop for i below entry-group-size
+                               for commit = (cl-git:next-revision walker)
+                               while (not (null commit))
+                               for shortlog = (first (split-sequence:split-sequence #\Newline (cl-git:message commit)))
```

Comment:
please move the cl-git logic either to porcelain or to another package. The legit command should be VCS-agnostic and the more about displaying as possible.

you also have `str:split` or `str:words`, choose your weapon :p